### DISCUSSION TITLE
Support for other render modes

### DISCUSSION THREAD

**Discussion by sweco-benebj** [egil/Htmxor#42] [Q&A] [answered]
Hello, I just want to make sure, but considering the overrides that ```AddHtmx``` extension method introduces under the hood to the rendering and routing it is probably out of question to support any other render mode than SSR, correct?

**Comment by egil** [comment, marked as answer]
It is not out of the question, it should be quite possible to support technically, but I have probably pulled out a bunch of bits in Htmxors copy of the original services in Blazor that enable switching render modes on the fly for it to "just work" right now. Putting those bits back in is the challenge, but I do not think there are anything in Htmxor that prevents it.

 If you mean that using other render modes together in pages that uses htmx, then probably no, and I don't see the point either.

**Reply by sweco-benebj** [reply]
Thanks for the clarification! Indeed, using interactive components in a htmxor page does not make much sense. I was just thinking about the scenario of gradually introducing htmxor pages into an app that already uses different render modes.

P.S. Thanks for the all effort that you're putting into the project, it's really an inspiring work!

**Reply by egil** [reply]
That's actually a "direction" I was not thinking of, and a really good point. If Htmxor can be made to be a drop-in replacement, without breaking existing functionality, then it's possible to do as you say, to gradually introduce htmx-enabled pages.

That makes it a very good reason for me to prioritize reenabling standard Blazor and have Htmxor be unobtrusive.
